### PR TITLE
Fixes #25091: Add VARIABLE_NAME in technique parameters so that we can use variable real name

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/25091

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/CfclerkXmlConstants.scala [hunk 1]
```diff
@@ -110,6 +110,7 @@ object CfclerkXmlConstants {
   val VAR_NAME                        = "NAME"
   val VAR_DESCRIPTION                 = "DESCRIPTION"
   val VAR_LONG_DESCRIPTION            = "LONGDESCRIPTION"
+  val VAR_VARIABLE_NAME               = "VARIABLE_NAME"
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/CfclerkXmlConstants.scala:113]: I would use `VARIABLENAME` to be consistent with other properties here. Or change for underscore everywhere.

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/VariableSpecParser.scala [hunk 2]
```diff
@@ -190,6 +191,12 @@ class VariableSpecParser extends Loggable {
                 )
               }
             }
+            variableName   <- (elt \ VAR_VARIABLE_NAME).toList match {
+                                case var_name :: Nil => Right(Some(var_name.text))
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/xmlparsers/VariableSpecParser.scala:195]: this is not conventional scala naming scheme, please use `varName`

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala [hunk 3]
```diff
@@ -117,14 +117,15 @@ object BuildBundleSequence {
...
     def quote(agentEscape: String => String): String
-    def name:  String
-    def value: String
+    def name:         String
+    def value:        String
+    def variableName: Option[String]
   }
   object BundleParam       {
-    final case class SimpleQuote(value: String, name: String) extends BundleParam {
+    final case class SimpleQuote(value: String, name: String, variableName: Option[String]) extends BundleParam {
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/BuildBundleSequence.scala:125]: why not always using it ? Is it important to use the ID on linux (I understand it is important to use the name on windows)

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/VariableAndSectionSpec.scala [hunk 4]
```diff
@@ -248,10 +250,10 @@ final case class TrackerVariableSpec(
   override type T = TrackerVariableSpec
   override type V = TrackerVariable
 
-  override val name:        String = TRACKINGKEY
-  override val description: String = "Variable which kept information about the policy"
-
-  override val checked: Boolean = false
+  override val name:         String         = TRACKINGKEY
+  override val description:  String         = "Variable which kept information about the policy"
+  override val variableName: Option[String] = None
```
Comment (fanf) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/VariableAndSectionSpec.scala:255]: there's something I don't get here: we have : 
- name
- description
- variableName
- longDescription
- id

But we only have in yaml / metadata: 
- id
- name
- description
- documentation

Mapping from/to longdescription <-> documentation is clear ; but I don't get the remaining ones. 

I'm almost sure `name` actually contains `id` (always). And then should be named `id`. 
I'm not sure about what contains `id: Option[String]` nor name. 
And I think there is a difference between a "display name" which is perhaps a description ?

**Comment by fanf** [comment]
I think I'm really missing a schema or document that list the names of all things at all level and evolution proposed here. Especially around usage of id/variable name/display name/description at different places. 

The categories I'm interested in are: 
- the yml technique
- the technique editor UI
- the metadata.xml and techniques (cf, ps1)
- the "parameters" element of technique in directive/technique API and json directive
- the directive UI
- the directive serialisation in LDAP
- the directive serialisation in XML

And not sure it matters (I think it's more variable value than name here):
- expected reports
- the resulting reports in db

And perhaps other things ?

Ideally, it would a unit test, but at least the part from yaml to metadata would be hard to do (perhaps other part are at the frontier of systems, but at least they are mostly in the same language)

File: policies/rudderc/src/backends/metadata.rs [hunk 5]
```diff
@@ -389,11 +392,12 @@ struct Input {
 struct SelectOne {
     // actually, the id
     name: String,
+    #[serde(rename = "VARIABLE_NAME")]
```
Comment (amousset) [inline policies/rudderc/src/backends/metadata.rs:395]:
```suggestion
```

No need, everything is already uppercased by line `#[serde(rename_all = "UPPERCASE")]` on the struct

File: policies/rudderc/src/backends/metadata.rs [hunk 6]
```diff
@@ -375,11 +377,12 @@ struct SectionInput {
 struct Input {
     // actually, the id
     name: String,
+    #[serde(rename = "VARIABLE_NAME")]
```
Comment (amousset) [inline policies/rudderc/src/backends/metadata.rs:380]:
```suggestion
```

**Comment by fanf** [comment]
Replaced by rudderc/src/backends/metadata.rs:523:56

**Pull request closed without merging**